streaming/base/sampling.py

```diff
@@ -23,6 +23,18 @@ def get_sampling(samples_per_shard: NDArray[np.int64], choose: int, granularity:
     Returns:
         NDArray[np.int64]: Array of ephemeral samples chosen per shard.
     """
+    if choose < 0:
```

Comment:
Why do we need this since we already have a check [here](https://github.com/mosaicml/streaming/blob/main/streaming/base/stream.py#L128) ?

Reply:
I am speculating that in the future, a user may want to do something custom that we didn't anticipate, resulting in them calling this method directly instead of indirectly through StreamingDataset.

In that case, it would be correct to validate all its arguments.

That may seem too speculative, but it's practically free to validate all args, so let's play it safe.

Reply:
Put another way, if a method exists in the repo, someone somewhere is going to call it (while expecting all args to be validated)